File: TypeScript/linear-search.ts
const arr = [15, 2, 1, 8, 10, 5, 20, 25];
const numberSought = 8;

const linearSearch = (numbers: number[], wanted: number) => {
    for (const number of numbers) 
        if (number === wanted) return true;

    return false;
}


linearSearch(arr, numberSought); 

File: Dart/linear-search.dart
void main(){
  List<int> arr = [5, 10, 15, 20, 25, 30];
  int x = 15;
  int result = linearSearch(arr, x);
  print(result);
}

linearSearch(List<int> array, int x){
  for(int i = 0; i< array.length; i++){
    if(array[i] == x){
      return i; //returns the position in the array.
    }
  }
}
